components/feature/recentlyclosed/src/main/java/mozilla/components/feature/recentlyclosed/adapter/RecentlyClosedTabAdapter.kt

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.recentlyclosed.adapter
+
+import android.content.Context
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.session.ext.readSnapshotItem
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.recentlyclosed.RecentlyClosedTab
+import mozilla.components.feature.recentlyclosed.db.RecentlyClosedTabEntity
+
+internal class RecentlyClosedTabAdapter(
+    val entity: RecentlyClosedTabEntity
+) : RecentlyClosedTab {
+    override val id: Long
+        get() = entity.id!!
+
+    override val title: String
+        get() = entity.title
+
+    override val url: String
+        get() = entity.url
+
+    override fun restoreRecentlyClosedTab(
+        context: Context,
+        engine: Engine,
+        restoreSessionId: Boolean
+    ): SessionManager.Snapshot.Item? {
+        return entity.getStateFile(context.filesDir)
+            .readSnapshotItem(engine, restoreSessionId, false)
```

Comment:
We should also decouple this from the `RecentlyClosedTabAdapter` entirely since this action doesn't belong here, if I understand correctly what we're trying to do.

What we probably want is something that converts this entity to a `TabSessionState` so we can call:
```kotlin
val tabsToRestore = listOf(tabsToRestore)
browserStore.dispatch(RestoreAction(tabsToRestore))
```

This might be easier than using the SessionManager since:
 1. The SessionManager is (almost) deprecated and we're trying to avoid writing new code with it.
 2. There isn't an easy to use API for the SessionManager that lets you do this today.

Reply:
From what I can tell the `SessionManager` and snapshots are currently the only way to restore new tabs fully? Dispatching the TabListAction.AddTabAction or TabListAction.RestoreAction won't hook up the engine session and just give a blank tab when I try them. I may be misunderstanding what I'm trying to do though. 🤔